### PR TITLE
Fix the ability to skip baseline sims in Docker-based implementations

### PR DESCRIPTION
**Pull request by lathanh**
## Checklist

Not all may apply

- [x] Code changes (must work)
- [ ] ~Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)~
- [x] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/coverage.yml` as necessary.
- [ ] All other unit and integration tests passing
- [ ] ~Update validation for project config yaml file changes~
- [ ] ~Update existing documentation~
- [ ] ~Run a small batch run on Kestrel/Eagle to make sure it all works if you made changes that will affect Kestrel/Eagle~
- [ ] ~Add to the changelog_dev.rst file and propose migration text in the pull request~

### CODE REVIEW

File: buildstockbatch/cloud/docker_base.py [hunk 1]
```diff
@@ -314,7 +314,10 @@ def _prep_jobs_for_batch(self, tmppath):
         self.validate_buildstock_csv(self.project_filename, df)
         building_ids = df.index.tolist()
         n_datapoints = len(building_ids)
-        n_sims = n_datapoints * (len(self.cfg.get("upgrades", [])) + 1)
+        if self.skip_baseline_sims:
```
Comment (mfathollahzadeh) [inline buildstockbatch/cloud/docker_base.py:317]: I think the key on buildstock batch documentation is skip_sims. I would suggest we stay consistent with that on keys.

Comment (mfathollahzadeh) [reply buildstockbatch/cloud/docker_base.py:317]: upon checking on schemas: 

sim-spec:
  n_buildings_represented: int(required=True)
  skip_sims: bool(required=False)
  custom_gems: bool(required=False)

**Review by mfathollahzadeh — COMMENTED** [review]
Thanks, Robert! This looks good to me and I just added a small comment

File: buildstockbatch/cloud/docker_base.py (hunk 1 above)
Comment (nweires) [reply buildstockbatch/cloud/docker_base.py:317]: This is fetching the value of `skip_sims` [here](https://github.com/rewiringamerica/buildstockbatch/blob/661f66abc1e78faf47c9ba56c07ba0918278a15c/buildstockbatch/base.py#L134).

File: buildstockbatch/cloud/docker_base.py [hunk 2]
```diff
@@ -327,9 +330,12 @@ def _prep_jobs_for_batch(self, tmppath):
...
 
         # Create list of (building ID, upgrade to apply) pairs for all simulations to run.
-        baseline_sims = zip(building_ids, itertools.repeat(None))
         upgrade_sims = itertools.product(building_ids, range(len(self.cfg.get("upgrades", []))))
-        all_sims = list(itertools.chain(baseline_sims, upgrade_sims))
+        if not self.skip_baseline_sims:
+            baseline_sims = zip(building_ids, itertools.repeat(None))
+            all_sims = list(itertools.chain(baseline_sims, upgrade_sims))
+        else:
+            all_sims = list(itertools.chain(upgrade_sims))
```
Comment (nweires) [inline buildstockbatch/cloud/docker_base.py]:
```suggestion
            all_sims = list(upgrade_sims)
```

**Review by nweires — COMMENTED** [review]
Can you also add a test case [here](https://github.com/rewiringamerica/buildstockbatch/blob/3cbb673e2389d0d21e36beb9bbbe7c891786db06/buildstockbatch/test/test_docker_base.py#L21) for this?

**Comment by lathanh** [comment]
> Can you also add a test case [here](https://github.com/rewiringamerica/buildstockbatch/blob/3cbb673e2389d0d21e36beb9bbbe7c891786db06/buildstockbatch/test/test_docker_base.py#L21) for this?

Done

File: buildstockbatch/test/test_docker_base.py [hunk 3]
```diff
@@ -18,6 +18,52 @@
 resources_dir = os.path.join(here, "test_inputs", "test_openstudio_buildstock", "resources")
 
 
+@docker_available
+def test_skip_baseline_sims_false(basic_residential_project_file, mocker):
```
Comment (nweires) [inline buildstockbatch/test/test_docker_base.py]: These could be combined into a [single parametrized test](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) with something like:
```suggestion
@pytest.mark.parametrize("skip_sims,expected_n_sims", [(False, 10), (True, 5)])
def test_skip_baseline_sims_false(basic_residential_project_file, mocker, skip_sims, expected_n_sims):
```

**Review by nweires — APPROVED** [review]

**Pull request merged**